jzaki/bls-wallet
# Monorepo setup

## What is this PR doing?
This PR adds a streamlined setup process and documentation for the monorepo of `bls-wallet` components. It does this by:
- Adding a `setup.ts` script for initial repo setup.
- Consolidating contract addresses and deployment metadata into a `network.json` file, which can be loaded/parsed with `NetworkConfig` type and functions in `clients`.

## How can these changes be manually tested?
- Clone a fresh copy of the repo and checkout this branch OR (WARNING this will clear all untracked files) checkout this branch and run `git clean -xdf`.
- Follow the setup instructions in the top level `README.md`
- Ensure tests run and pass for `./aggregator`.
- Ensure tests run and pass for `./contracts`.
- With contracts deployed, aggregator running, and extension added to a browser, ensure a new BLS wallet can be created and a txn sent.

## Does this PR resolve or contribute to any issues?
Resolves https://github.com/jzaki/bls-wallet/issues/4

## Checklist

## Guidelines
- If your PR is not ready, mark it as a draft
- The `resolve conversation` button is for reviewers, not authors
  - (But add a 'done' comment or similar)

### TODO
- [x] Move contract addresses and other deployments to network JSON file (if agreed that it is a good idea)
- [x] `docker-compose` file ~to easily start all services~
- [x] Setup script
~- [ ] Replace remaining usages of `VERIFICATION_GATEWAY_ADDRESS` in `./extension`~ Ran into some config issues getting webpack to properly read in deployed contracts config at build time, will need to revisit.
- [x] Complete main `README`s
- [x] Update sub `README`s
- [x] Add local depoy setup and documentation.
